### PR TITLE
llvm37 3.7.0 (new formula)

### PR DESCRIPTION
**Pull request by lembacon**

### CODE REVIEW

**Comment by jenshnielsen** [comment]
Does this support OpenMP out of the box or does it need to depend on libiomp like the clang-omp formula?

**Comment by chfast** [comment]
Any chances it is merged?

**Comment by zeux** [comment]
@jenshnielsen When I try to build something with -fopenmp switch with clang from this formula, I get:

    ~ $ clang++-3.7 test.cpp -fopenmp
    ld: library not found for -lgomp

Installing libiomp does not fix this though. I'm not sure what the official status of OpenMP support is on OSX; note that I get the same linker error if I use clang from Xcode 7 with -fopenmp switch.

File: llvm37.rb [hunk 1]
```diff
@@ -0,0 +1,288 @@
+class CodesignRequirement < Requirement
+  include FileUtils
+  fatal true
+
+  satisfy(:build_env => false) do
```
Comment (DomT4) [inline llvm37.rb]: Can you tweak this block a bit to match:

```ruby
  satisfy(:build_env => false) do
    mktemp do
      touch "llvm_check.txt"
      quiet_system "/usr/bin/codesign", "-s", "lldb_codesign", "--dryrun", "llvm_check.txt"
    end
  end
```
The `"--dryrun"` flag stops the check actually modifying anything on the filesystem.

**Comment by chfast** [comment]
Ping.

**Comment by DomT4** [comment]
@BrewTestBot test this please

File: llvm37.rb (hunk 1 above)
Comment (lembacon) [inline llvm37.rb]: Changed via https://github.com/lembacon/homebrew-versions/commit/e00bde350fbd2c4ef8e3dd916dd8565d7e70590c.

**Comment by chfast** [comment]
Does it need any more work?

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by DomT4** [comment]
Going to merge this because people keep asking and LLVM haven't rushed a new release, but please do bear in mind the accidentally broken API element if you run into unusual breakage against local projects. There's not really a lot we can or will do to resolve that until LLVM release the fix upstream.

Merged in https://github.com/Homebrew/homebrew-versions/commit/4cff8a0ca4c67d31d25177165429a5157bb09cb6. Thank you for another contribution to Homebrew @lembacon! :smiley_cat:

**Comment by danilaml** [comment]
@DomT4 any help with https://github.com/Homebrew/homebrew-versions/issues/1076 ?

**Comment by DomT4** [comment]
Absolutely no idea I'm afraid. I'll try and take a look but LLVM expert I am definitely not, to say the least.

**Comment by danilaml** [comment]
@DomT4 who created the bottle though?

**Comment by DomT4** [comment]
The CI machine. Asking him/her/it won't get you much of a reply, I'm afraid.

**Comment by DomT4** [comment]
It's worth checking if your error reproduces from source. `brew reinstall llvm37 -s`. Obviously, bear in mind LLVM takes hours to build.

**Comment by danilaml** [comment]
@DomT4 thats no the problem, Homebrew is needed for me only as provider of precompiled packages to not build llvm itself (I use it for travis ci). CMake modules of LLVM are generated after compiling (for example in old 3.6 bottles you can see user specific past to llvm lib in them) but in 3.7 bottle some parts are undefined and comment is left so I assume that the problem is with however someone made such bottle.

**Comment by danilaml** [comment]
@DomT4 
I compared to the cmake generated file on my windows machine (but it should be the same on OSX) and I think I found the issue. I have:
```
# Compute the installation prefix from this LLVMConfig.cmake file location.
get_filename_component(LLVM_INSTALL_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
get_filename_component(LLVM_INSTALL_PREFIX "${LLVM_INSTALL_PREFIX}" PATH)
get_filename_component(LLVM_INSTALL_PREFIX "${LLVM_INSTALL_PREFIX}" PATH)
get_filename_component(LLVM_INSTALL_PREFIX "${LLVM_INSTALL_PREFIX}" PATH)
```
But in Homebrew bottle it's:
```
# Compute the CMake directory from the LLVMConfig.cmake file location.n get_filename_component(_LLVM_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)n # Compute the installation prefix from the LLVMConfig.cmake file location.n get_filename_component(LLVM_INSTALL_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)n get_filename_component(LLVM_INSTALL_PREFIX "${LLVM_INSTALL_PREFIX}" PATH)n get_filename_component(LLVM_INSTALL_PREFIX "${LLVM_INSTALL_PREFIX}" PATH)n get_filename_component(LLVM_INSTALL_PREFIX "${LLVM_INSTALL_PREFIX}" PATH)n set(_LLVM_LIBRARY_DIR "${LLVM_INSTALL_PREFIX}/lib")
```

Notice the difference? For some reason on OSX it's just `n` instead of newline (`\n`). I'm not sure if this one specific build issue, bug in cmake installed on travis ci or bug in how llvm handles osx.

Could you rebuild the bottle (maybe with updated cmake) and see if it helps? If possible the 3.7.1 llvm version (it's been tagged last week but not yet officially released across all platforms if I got that correctly: http://lists.llvm.org/pipermail/llvm-dev/2015-December/093179.html )

**Comment by mikemcquaid** [comment]
@danilaml You can try locally rebuilding using `--build-from-source`

**Pull request closed without merging**